q.js

```diff
@@ -1540,6 +1540,56 @@ Promise.prototype.all = function () {
 };
 
 /**
+ * Returns the first resolved promise of an array. Prior rejected promises are
+ * ignored.  Rejects only if all promises are rejected.
+ * @param {Array*} an array containing values or promises for values
+ * @returns a promise fulfilled with the value of the first resolved promise,
+ * or a rejected promise if all promises are rejected.
+ */
+Q.any = any;
+
+function any(promises) {
+    var deferred = Q.defer();
+    if (promises.length === 0) {
+        return Q.resolve();
+    }
+
+    var promisesCount = promises.length;
+    for (var i = 0; i < promises.length; i++) {
+        var promise = promises[i];
+        /* Let's define that false could be a value to resolve with, so let's not
+         * if(!promise)
+         */
+        if (promise === undefined || promise === null) {
+            promisesCount--;
+            continue;
+        }
+
+        (function(promise, i) {
+            Q.resolve(promise).then(function(result) {
+                deferred.resolve(result);
+            }, function(reason) {
+                promisesCount--;
+                if(promisesCount === 0) {
+                    deferred.reject(new Error("No promises resolved successfully."));
```

Comment:
This might be more expressive as "Can't get fulfillment value from any promise, all promises were rejected."